workflow/internal/dispatcher/status.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dispatcher
+
+import (
+	"fmt"
+)
+
+type TaskState int8
+
+const (
+	PENDING TaskState = iota
+	RUNNING
+	DONE
+	FAILED
+)
+
+func (s TaskState) String() string {
+	switch s {
+	case PENDING:
+		return "Pending"
+	case RUNNING:
+		return "Running"
+	case DONE:
+		return "Done"
+	case FAILED:
+		return "Failed"
+	default:
+		return "Unknown"
+	}
+}
+
+type TaskStatus struct {
+	id    string
```

Comment:
wdy if we remove the ID from the task status? as i noticed, we use this in `Dispatcher.set` and `Dispatcher.update`, but i think it might be cleaner to explicity pass the id to those two methods than infer from the status data. `TaskStatus.String` is not used anywhere where we dont already have ID anyway.

Reply:
Originally I used that approach. I'll revert to that.